app/controllers/telegram_webhooks_controller.rb

```diff
@@ -0,0 +1,119 @@
+class TelegramWebhooksController < Telegram::Bot::UpdatesController
+  include Telegram::Bot::UpdatesController::MessageContext
+  include Telegram::Bot::UpdatesController::CallbackQueryContext
+
+  def start!(*)
+    if session?
+      respond_with :message, text: 'Choose action', reply_markup: menu_markup
+    else
+      save_context :auth
+      respond_with :message, text: 'Hi! Give me your mobile number.', reply_markup: phone_markup
```

Comment:
maybe there is the ability to check user's phone number throughout telegram. I mean you can prompt: "Login with my phone number or enter another one" if user's phone number is available or just "give me your number" if it's not. That would be a cool feature

Reply:
If I give posibility enter anothe phone number, user will get access for data that isn't his.

Reply:
i think its bad idea to check by phone number i think each person should have his own access key which he can use to get access to information

Reply:
Anton said me to use phone number. Use phone number is one of quick methods login and send meter reading.

Reply:
Yep i think auth by number is ok as telegram has it own strong authentication
so there is no need to overthink this auth step in scope of a educational project